jsc8 audit changes for StreamWorkers (#273)
Macrometacorp/docs: docs/cep/stream-workers-example.md

    console.log("--- Updating Stream Application `Sample-Cargo-App`");
    result = await app.updateApplication([], updatedAppDefinition);
```    
  </TabItem>
</Tabs>

> ## Run an Adhoc Query

Reviewer: ```suggestion
## Run an Ad Hoc Query
```

Reviewer: "ad hoc" is always two words.

> In this example, we run an adhoc query on the store `SampleCargoAppDestTable` used in Stream Application. It should get records which you inserted into `SampleCargoAppInputTable`.

Reviewer: ```suggestion
In this example, we run an ad hoc query on the store `SampleCargoAppDestTable` used in a stream application. It should get records which you inserted into `SampleCargoAppInputTable`.
```

```js

>     console.log("--- Deleting Stream Application `Sample-Cargo-App`");

Reviewer: ```suggestion
    console.log("--- Deleting stream application `Sample-Cargo-App`");
```

  <TabItem value="js" label="Javascript">

```js
const jsc8 = require("jsc8");

// Variables
const streamAppName = "Sample-Cargo-App";
const sourceCollectionName = "SampleCargoAppInputTable";
const destinationCollectionName = "SampleCargoAppDestTable";
const parameter = { weight: "" };
const globalUrl = "https://gdn.paas.macrometa.io";
const thisApiKey = "XXXXX";

const insertDataValue = {
  query: {
    name: "insertWeight",
    value: `INSERT { weight:@weight } IN ${sourceCollectionName}`,
    parameter
  }
};

const queryName = insertDataValue.query.name.toString();
const queryValue = insertDataValue.query.value.toString();

// App definitions
const appDefinition =
  `@App:name('Sample-Cargo-App')
  @App:qlVersion("2")

>   @App:description('Basic Stream application to demonstrate reading data from input stream and store it in the collection. The stream and collections will be created automatically if they do not already exist.')

Reviewer: ```suggestion
  @App:description('Basic stream application to demonstrate reading data from input stream and store it in the collection. The stream and collections will be created automatically if they do not already exist.')
```

  CREATE SOURCE SampleCargoAppInputTable WITH (type = 'database', collection = "SampleCargoAppInputTable", collection.type="doc", replication.type="global", map.type='json') (weight int);

  -- Create Stream SampleCargoAppDestStream
  CREATE SINK SampleCargoAppDestStream WITH (type = 'stream', stream = "SampleCargoAppDestStream", replication.type="local") (weight int);

  CREATE STORE SampleCargoAppDestTable WITH (type = 'database', stream = "SampleCargoAppDestTable") (weight int);

  -- Data Processing
  @info(name='Query')
  INSERT INTO SampleCargoAppDestStream
  SELECT weight
  FROM SampleCargoAppInputTable;

  -- Data Processing
  @info(name='Dump')
  INSERT INTO SampleCargoAppDestTable
  SELECT weight
  FROM SampleCargoAppInputTable;`;

// This is a supporting function
function messageHandler (error) {
  const message = {
    "StatusCode ": error.statusCode,
    "ErrorMessage ": error.message,
    "ErrorNum ": error.errorNum
  };
  console.log(message);
}

const sleep = (milliseconds) => {
  return new Promise(resolve => setTimeout(resolve, milliseconds));
};

async function main () {

  console.log("--- Connecting to GDN");
  // email = "";
  // password = "";

>   // Create an authenticated instance with token or API key

Reviewer: ```suggestion
  // Create an authenticated instance with a token or API key
```

  // const client = new jsc8({url: "https://gdn.paas.macrometa.io", token: "XXXX", fabricName: '_system'});
  const client = new jsc8({ url: globalUrl, apiKey: thisApiKey, fabricName: "_system" });
  // console.log("Authentication done!!...");
  

>   // Or use Email & Password to Authenticate client instance

Reviewer: ```suggestion
  // Or use email and password to authenticate a client instance
```

  // const client = new jsc8("https://gdn.paas.macrometa.io");
  // await client.login("email", "password");

>   console.log("Connecttion successful");

Reviewer: ```suggestion
  console.log("Connection successful");
```

    result = await client.getStreamApp(streamAppName);
    console.log(result);
  }

  async function enablingStreamApplication () {
    console.log("--- Enable stream application " + streamAppName);
    // Enable app using change_state function
    const result = await client.activateStreamApp(streamAppName, true);
    console.log(result);
    console.log("The connection has been opened.");
  }

  async function updatingSteamApplication () {
    const app = await client.streamApp(streamAppName);

    console.log("--- Updating stream application " + streamAppName);
    await app.updateApplication([], updatedAppDefinition);
    // Enable app using change_state function
    await app.activateStreamApplication(true);
  }

  async function insertingData () {
    console.log("Waiting for 30 seconds for all the resources to be ready");
    await sleep(30000);
    console.log("--- Inserting data to " + sourceCollectionName);
    for (i = 1; i <= 50; i++) {
      await client.executeRestql(queryName, { weight: i });
      console.log(i);
    }
  }

  async function disablingStreamApplication () {
    await client.activateStreamApp(streamAppName, false);

>     console.log("The connection has been closed.");

Reviewer: This doc is inconsistent with whether periods are used or not in the log. Please choose one and stick with it. Personally, I prefer the periods at the end of sentences, but I am not sure what is normal in JS logs.

Author: Done. I have actually removed the periods since the periods were mostly present in the logs added by us and the existing file logs didn't have periods it seems

  }

  async function displayingResults () {
    const app = await client.streamApp(streamAppName);

    // As per the query, the result is limited to the first 10 results
    console.log(`--- Running adhoc query on the store ${destinationCollectionName} used in stream application. 
      It should get all records which you inserted into ${sourceCollectionName}.`);
    const q = `select * from ${destinationCollectionName} limit 10`;
    const result = await app.query(q);
    console.log(result);
  }

  async function deletingStreamApplication () {
    const app = await client.streamApp(streamAppName);

>     console.log(`--- Deleting Stream application ${streamAppName}`);

Reviewer: ```suggestion
    console.log(`--- Deleting stream application ${streamAppName}`);
```

    await app.deleteApplication();
  }

  async function displayingSampleApplications () {
    console.log("--- You can try out several stream applications which are pre-loaded and ready to run.");
    const result = await client.getStreamAppSamples();
    console.log("Sample Stream Applications:");
    console.log(result);
  }

  async function run () {
    try {
      await creatingQueryWorker();
      await creatingStreamApplication();
      await enablingStreamApplication();
      await updatingSteamApplication();
      await insertingData();
      await disablingStreamApplication();
      await displayingResults();

>       // Only the Stream Application is being automatically removed with the code

Reviewer: ```suggestion
      // Only the stream application is removed with the code
```
